task_templates/pipelines/python3_xgboost/custom.py

```diff
@@ -16,144 +24,103 @@ def fit(
+    row_weights: Optional[np.ndarray]
         An array of non-negative numeric values which can be used to dictate how important
         a row is. Row weights is only optionally used, and there will be no filtering for which
         custom models support this. There are two situations when values will be passed into
-        row_weights, during smart downsampling and when weights are explicitly provided by the user
+        row_weights, during smart downsampling and when weights are explicitly specified in the project settings.
     kwargs
-        Added for forwards compatibility
+        Added for forwards compatibility.
 
     Returns
     -------
-    Nothing
+    None
+        fit() doesn't return anything, but must output an artifact
+        (typically containing a trained object) into output_dir
+        so that the trained object can be used during scoring.
     """
-    # Feel free to delete which ever one of these you aren't using
     if class_order:
         estimator = make_classifier_pipeline(X, len(class_order))
-        if y.dtype == np.dtype("bool"):
-            y = y.astype("str")
         lb = LabelEncoder()
         y = lb.fit_transform(y)
     else:
         estimator = make_regressor_pipeline(X)
     estimator.fit(X, y)
 
-    # NOTE: We currently set a 10GB limit to the size of the serialized model
+    # Dump the trained object [in this example - a trained XGBoost model]
+    # into an artifact [in this example - artifact.pkl]
+    # and save it into output_dir so that it can be used later when scoring data
+    # Note: DRUM will automatically load the model when it is in the default format (see docs)
+    # and there is only one artifact file
     output_dir_path = Path(output_dir)
     if output_dir_path.exists() and output_dir_path.is_dir():
         with open("{}/artifact.pkl".format(output_dir), "wb") as fp:
             pickle.dump(estimator, fp)
 
 
-"""
-Custom hooks for prediction
----------------------------
+def score(data: pd.DataFrame, model: Any, **kwargs: Dict[str, Any]) -> pd.DataFrame:
+    """
+    DataRobot will run this hook when the task is used for scoring inside a blueprint
 
-If drum's standard assumptions are incorrect for your model, DRUM supports several hooks
-for custom inference code.
-"""
-# def init(code_dir : Optional[str], **kwargs) -> None:
-#     """
-#
-#     Parameters
-#     ----------
-#     code_dir : code folder passed in the `--code_dir` parameter
-#     kwargs : future proofing
-#     """
-
-# def load_model(code_dir: str) -> Any:
-#     """
-#     Can be used to load supported models if your model has multiple artifacts, or for loading
-#     models that DRUM does not natively support
-#
-#     Parameters
-#     ----------
-#     code_dir : is the directory where model artifact and additional code are provided, passed in
-#
-#     Returns
-#     -------
-#     If used, this hook must return a non-None value
```

Comment:
@4dahalibut , I deleted this after conversations with @MikhailYakubovskiy because having commented out hooks seemed confusing to the user (particularly since one of them, init, is mainly for R models anyway).

Reply:
Did misha talk to cfdses about it? is it in fact confusing?

Reply:
@MikhailYakubovskiy ?